src/main/java/org/spongepowered/api/item/inventory/ItemStackComparators.java

```diff
@@ -82,46 +69,100 @@
     public static final Comparator<ItemStack> TYPE_SIZE = Ordering.compound(Arrays.asList(TYPE, SIZE));
 
     /**
-     * Compares ItemStacks based on {@link org.spongepowered.api.item.ItemType},
-     * {@link ItemStack} damage and {@link ItemStack} size. This comparator will
-     * not return the same results as ItemStack.equals(ItemStack) for ItemStacks
-     * with extra attached data.
+     * The default comparator for {@link ItemStack}s.
      */
-    @SuppressWarnings("unchecked")
-    public static final Comparator<ItemStack> TYPE_DAMAGE_SIZE = Ordering.compound(Arrays.asList(TYPE, DAMAGE, SIZE));
+    public static final Comparator<ItemStack> DEFAULT = TYPE_SIZE;
 
-    /**
-     * The default comparator for {@link org.spongepowered.api.item.inventory.ItemStack}s.
-     */
-    public static final Comparator<ItemStack> DEFAULT = TYPE_DAMAGE_SIZE;
+    public static final Comparator<ItemStack> PROPERTIES = new Properties();
+
+    public static final Comparator<ItemStack> ITEM_DATA = new ItemDataComparator();
+
+    public static final Comparator<ItemStack> ALL = Ordering.compound(Arrays.asList(TYPE, SIZE, PROPERTIES, ITEM_DATA));
```

Comment:
My comment was not about to remove the all comparator. Only PROPERTIES bugged me a little bit.
I do not know tinkersconstructtools but for me this does not look like an ItemProperty and more like an ItemData.